src/Shared/Microsoft.DotNet.Kusto/IKustoClientProvider.cs

```diff
@@ -4,11 +4,14 @@
 
 using System.Threading.Tasks;
 using System.Data;
+using Kusto.Data.Results;
 
 namespace Microsoft.DotNet.Kusto
 {
     public interface IKustoClientProvider
     {
         Task<IDataReader> ExecuteKustoQueryAsync(KustoQuery query);
+
+        Task<ProgressiveDataSet> ExecuteStreamableKustoQueryAsync(KustoQuery query);
```

Comment:
I think we should probably not return this nutso object to the callers, and wrap it somehow.  (Usage of it involves a LOT of switch statements and such).  Maybe just an IEnumerable<IDataReader>() would work?  Or even an IDataReader implementation that wraps them all together if you are feeling really fancy.

```c#
public MyDataReader : IDataReader {
  IEnumerable<Frame> _set;
  IDataReader _current;

  MoveNext(){
    while (!_current.MoveNext()) {
     if (_set.MoveNext()) {
       _current = GetDataReaderFromFrameIfItHasOne();
     }
   }
  }
```

Reply:
Bleh. This type is _horrible_ it _might_ return "DataTable" frame, which is just a silly IDataReader. Or it might return a DataTableFragment" frame, which returns row by row of "object[]" (so you need to know what types to coerce things into).

I'm not sure why it would return one v the other, which is probably why I never bothered with this API.

Wrapping that all in something... sane... is going to be spicy.

Reply:
Also: None of it is async for some strange reason.